cmake/CMakeLists.txt

```diff
@@ -81,13 +81,13 @@ option(onnxruntime_USE_AVX2 "Use AVX2 instructions" OFF)
 option(onnxruntime_USE_AVX512 "Use AVX512 instructions" OFF)
 
 option(onnxruntime_USE_OPENMP "Build with OpenMP support" OFF)
-option(onnxruntime_BUILD_SHARED_LIB "Build a shared library" OFF)
+option(onnxruntime_BUILD_SHARED_LIB "Build a shared library" ON)
```

Comment:
You don't need to modify the default values of these options. They are always be set or override over command line.  For example: "-Donnxruntime_BUILD_SHARED_LIB=ON".